modules/plot/main/coffee/Axis.coffee

```diff
@@ -573,14 +530,69 @@ class Axis
       xStack
     else Math.max(start, 0)
 
-  # gets the height of the stack for a particular type, group and x value (the seriesId is the series that you want to get the baseline for)
   getYStack: (type, group, x, seriesId, start = 0) ->
+    allSeries = @series()
+    xscaletype = @x.scaleType()
+    yscaledomainmin = @yScale.domainMin
+
     if group
-      yStack = Math.max(@yScale.domainMin, 0)
-      for series, j in @series()
+      yStack = Math.max(yscaledomainmin, 0)
+      maybeys = allSeries.map (series) ->
         if series._.seriesId < seriesId and series.group() == group and series._.type == type
-          ys = series.getY(x, @x.scaleType() is 'discrete')
-          if hx.defined(ys)
-            yStack += ys
-      yStack
-    else Math.max(start, 0)
+          series.getY(x, xscaletype is 'discrete')
+      yStack + hx.sum maybeys.filter hx.identity
+    else
+      Math.max(start, 0)
+
+  calculateYBounds: ->
+    # The series must already be tagged before calling this method
+    # XXX: band series?
+    allSeries = @series()
+    xscaletype = @x.scaleType()
+    yscaledomainmin = @yScale.domainMin
+
+    initValue = { ymin: 0, ymax: 0 }
+    types = hx.groupBy(allSeries, (d) -> d._.type)
+    stackGroups = types.map ([type, series]) ->
+      type: type
+      group: hx.groupBy series, (s) -> if supportsGroup(s) then s.group() else undefined
+  
+  
+    typeGroupReductor = (type) =>
+      ({ ymin, ymax }, [seriesGroup, series]) =>
+        { yymin, yymax } = if seriesGroup == undefined
```

Comment:
again previously we were not actually using seriesGroup beyond checking if it was defined or not

now we use the value we already calculated rather than calculating one from topSeries and assuming it is representative